src/jtag3.c

```diff
@@ -2496,32 +2419,28 @@ void jtag3_print_parms1(const PROGRAMMER *pgm, const char *p) {
     return;
 
   if (b2_to_u16(buf) > 0) {
-    avrdude_message(MSG_INFO, "%sJTAG clock megaAVR/program   : %u kHz\n", p,
-      b2_to_u16(buf));
+    msg_info("%sJTAG clock megaAVR/program   : %u kHz\n", p, b2_to_u16(buf));
```

Comment:
would it make sense to use `imsg_level` here instead of `msg_level` and print the indent `p`?

Reply:
Thanks! I'll go through the PR in more detail later to search for potential errors.

I'm not a regex expert at all, so what would be the best pattern to find all lines that uses the "msg" functions and starts with `("%s`?

For instance, `imsg_info("%s` and `msg_error("%s` should be a match, but `pmsg_info("Hello` should not.

Reply:
`grep -r 'msg_[a-z2_]* *( *"%[0-9.-]*s' ` will do the trick, probably even the simpler `grep -r 'msg_[a-z2_]*("%s'`.

Well spotted, btw. This might make it unnecessary to pass `progbuf` to the `avr_display()`, `programmer_display()` and all the `pgm->display()` and the `pgm->print_parm1()` functions (though it would be a lot of work, given the sheer number of programmers, to change the API for these once the `progbuf` argument is no longer used or needed). See `grep -r progbuf`.

Reply:
It seems to me that only `ft245.c` and `jtag3.c` are using `progbuf`
```
$ grep -r progbuf
src/avrdude.h:extern char progbuf[];       // spaces same length as progname
src/ft245r.c:        pgm->display(pgm, progbuf);
src/jtag3.c:  jtag3_print_parms1(pgm, progbuf);
src/main.c:char   progbuf[PATH_MAX]; /* temporary buffer of spaces the same
src/main.c:    progbuf[i] = ' ';
src/main.c:  progbuf[i] = 0;
src/main.c:    avr_display(stderr, p, progbuf, verbose);
src/main.c:    programmer_display(pgm, progbuf);

```

Reply:
I've taken a closer look at the remaining code that still uses progbuf, and it seems to me like there's no quick fix here. Maybe this can be done in a separate PR sometime, but right now I think this PR is good as it is. I'll continue testing for a few days to see if I can spot messages with errors. But all in all, another excellent PR form @stefanrueger!